types/react-native/index.d.ts

```diff
@@ -3410,6 +3410,166 @@ export interface ImageStatic extends NativeMethodsMixin, React.ComponentClass<Im
 }
 
 /**
+ * @see https://facebook.github.io/react-native/docs/flatlist.html#props
+ */
+export interface FlatListProperties extends React.Props<FlatListStatic> {
+
+    /**
+     * Rendered in between each item, but not at the top or bottom
+     */
+    ItemSeparatorComponent?: React.ComponentClass<any>
```

Comment:
Looking at [the code](https://github.com/facebook/react-native/blob/v0.43.1/Libraries/CustomComponents/Lists/FlatList.js#L72), these are both _optional_ and _nullable_ (the leading question mark). So this should be:

```ts
ItemSeparatorComponent?: React.ComponentClass<any> | null
```